route/build/src/kwt_route.f90

```diff
@@ -408,216 +409,218 @@ SUBROUTINE QROUTE_RCH(IENS,JRCH,    & ! input: array indices
+    T_END   = T1 - (T1 - T0)*ROFFSET
+    allocate(FROUTE(0:NQ1),STAT=IERR)
+    if(ierr/=0)then; message=trim(message)//'problem allocating space for FROUTE'; return; endif
+    FROUTE(0) = .TRUE.; FROUTE(1:NQ1)=.FALSE.  ! init. routing flags
+    ! route flow through the current [JRCH] river segment (Q_JRCH in units of m2/s)
+    call KINWAV_RCH(JRCH,T_START,T_END,Q_JRCH(1:NQ1),TENTRY(1:NQ1),&     ! (input)
+                                       FROUTE(1:NQ1),T_EXIT(1:NQ1),NQ2,& ! (output)
+                                       ierr,cmessage)                    ! (output)
+    if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+    if(JRCH == ixPrint)then
+      print*, 'Q_JRCH = ', Q_JRCH
+      print*, 'FROUTE = ', FROUTE
+      print*, 'TENTRY = ', TENTRY
+      print*, 'T_EXIT = ', T_EXIT
+    endif 
+    ! ----------------------------------------------------------------------------------------
+    ! (4) COMPUTE TIME-STEP AVERAGES
+    ! ----------------------------------------------------------------------------------------
+    NR = COUNT(FROUTE)-1   ! -1 because of the zero element (last routed)
+    NN = NQ2-NR            ! number of non-routed points
+    TNEW = (/T_START,T_END/)
+    ! (zero position last routed; use of NR+1 instead of NR keeps next expected routed point) 
+    call INTERP_RCH(T_EXIT(0:NR+1),Q_JRCH(0:NR+1),TNEW,QNEW,IERR,CMESSAGE)
+    if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+    if(JRCH == ixPrint) print*, 'QNEW(1) = ', QNEW(1)
+    ! m2/s --> m3/s + instantaneous runoff from basin
+    RCHFLX(IENS,JRCH)%REACH_Q = QNEW(1)*RPARAM(JRCH)%R_WIDTH + RCHFLX(IENS,JRCH)%BASIN_QR(1)
+    ! ----------------------------------------------------------------------------------------
+    ! (5) HOUSEKEEPING
+    ! ----------------------------------------------------------------------------------------
+    ! compute the instantaneous flow at the end of the time step
+    !   (last routed point)
+    Q_END = Q_JRCH(NR) + &   !        (dQ/dT)                                 (dT)
+             ( (Q_JRCH(NR+1)-Q_JRCH(NR)) / (T_EXIT(NR+1)-T_EXIT(NR)) ) * (T_END-T_EXIT(NR))
+    ! compute an approximate entry time (needed for the remove routine later)
+    TIMEI = TENTRY(NR) + &   !        (dT/dT)                                 (dT)
+             ( (TENTRY(NR+1)-TENTRY(NR)) / (T_EXIT(NR+1)-T_EXIT(NR)) ) * (T_END-T_EXIT(NR))
+    ! allocate space for the routed data (+1 to allocate space for the interpolated point)
+    if (.not.allocated(KROUTE(IENS,JRCH)%KWAVE)) then
+      ierr=20; message=trim(message)//'KROUTE is not associated'; return
+    else
+   ! if(allocated(KROUTE(IENS,JRCH)%KWAVE)) then
+      deallocate(KROUTE(IENS,JRCH)%KWAVE, STAT=ierr)
+   !   if (ierr.ne.0) then
+   !     print *, '% CANNOT DEALLOCATE SPACE (FOR SOME UNKNOWN REASON)... TRY AND NULLIFY!'
+   !     !NULLIFY(KROUTE(IENS,JRCH)%KWAVE)
+   !     deallocate(KROUTE(IENS,JRCH)%KWAVE)
+   !   endif
+   ! endif
```

Comment:
remove commented out code

Reply:
Note Line 553-566 are updated.   I did not change logic.   I can clean up.